state/indexer/block/kv/kv.go

```diff
@@ -437,8 +512,9 @@ func (idx *BlockerIndexer) match(
 
 	// Remove/reduce matches in filteredHeights that were not found in this
 	// match (tmpHeights).
-	for k := range filteredHeights {
-		if tmpHeights[k] == nil {
+	for k, v := range filteredHeights {
+		tmpHeight := tmpHeights[k]
+		if (tmpHeight != nil && !bytes.Equal(tmpHeights[k], v)) || tmpHeight == nil {
```

Comment:
```suggestion
		if (tmpHeight != nil && !bytes.Equal(tmpHeight, v)) || tmpHeight == nil {
```

Reply:
... actually, since Golang implements short circuit evaluation for `||`, you can simplify this condition like so:
```go
tmpHeight == nil || !bytes.Equal(tmpHeight, v)
```